src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private int defaultSize = 10;
+    private int maxSize = 50;
+    private int maxBound = 5;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private int index = random.nextInt(maxBound);
```

Comment:
The `index` is calculated once during the instantiation of `FigureSupplier`. This means `getRandomFigure` will always return the same type of figure. Consider recalculating `index` inside the `getRandomFigure` method to ensure a different figure type can be returned each time.